lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1190,15 +1191,16 @@ def set_proj_type(self, proj_type, focal_length=None):
                                  f"None for proj_type = {proj_type}")
             self._focal_length = np.inf
 
-    def _roll_to_vertical(self, arr):
+    def _roll_to_vertical(self, arr, sign=1):
         """Roll arrays to match the different vertical axis."""
-        return np.roll(arr, self._vertical_axis - 2)
+        return np.roll(arr, sign * (self._vertical_axis - 2))
 
     def get_proj(self):
         """Create the projection matrix from the current viewing position."""
 
         # Transform to uniform world coordinates 0-1, 0-1, 0-1
         box_aspect = self._roll_to_vertical(self._box_aspect)
+        # self._box_aspect = box_aspect
```

Comment:
What's the reason for removing this line? I'm not totally clear where it comes into play (when rotating the view?), but note that the 3D test coverage isn't amazing so we should understand where behavior is changing.

Reply:
It was never there in main. I was just testing different ideas here.

Conclusion here is that box_aspect has to be local, otherwise there will be a new vertical axis each click.